raiden/smart_contracts/ChannelManagerLibrary.sol

```diff
@@ -37,6 +38,28 @@ library ChannelManagerLibrary {
     /// @return channel (address) the address of the NettingChannelContract of the two parties.
     function getChannelWith(Data storage self, address caller_address, address partner)
         constant
+        returns (address, bool)
+    {
+        bytes32 party_hash = partyHash(caller_address, partner);
+        if (self.channel_addresses[party_hash] == 0x0 || 
+            !contractExists(self, self.channel_addresses[party_hash])) {
```

Comment:
Hmmm can't we delete the channel once it has been settled by calling [`deleteChannel()`](https://github.com/raiden-network/raiden/blob/690df123b442a621d169682502fb7b1c77d57c92/raiden/smart_contracts/ChannelManagerContract.sol#L146)  ?